File: android/app/src/main/java/abi21_0_0/host/exp/exponent/modules/api/components/gesturehandler/react/RNGestureHandlerEventDataExtractor.java
package abi21_0_0.host.exp.exponent.modules.api.components.gesturehandler.react;

import abi21_0_0.com.facebook.react.bridge.WritableMap;
import abi21_0_0.host.exp.exponent.modules.api.components.gesturehandler.GestureHandler;

public interface RNGestureHandlerEventDataExtractor<T extends GestureHandler> {
  void extractEventData(T handler, WritableMap eventData);
}

File: android/vendored/sdk47/react-native-gesture-handler/android/src/main/java/abi47_0_0/com/swmansion/gesturehandler/react/RNGestureHandlerEventDataExtractor.kt
package abi47_0_0.com.swmansion.gesturehandler.react

import abi47_0_0.com.facebook.react.bridge.WritableMap
import abi47_0_0.com.swmansion.gesturehandler.core.GestureHandler

interface RNGestureHandlerEventDataExtractor<T : GestureHandler<T>> {
  fun extractEventData(handler: T, eventData: WritableMap)
}
